homeassistant/components/airnow/config_flow.py

```diff
@@ -0,0 +1,114 @@
+"""Config flow for AirNow integration."""
+import logging
+
+from pyairnow import WebServiceAPI
+from pyairnow.errors import AirNowError, InvalidKeyError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_RADIUS,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_NAME, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """
+    Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    session = async_get_clientsession(hass)
+    client = WebServiceAPI(data[CONF_API_KEY], session=session)
+
+    lat = data[CONF_LATITUDE]
+    lng = data[CONF_LONGITUDE]
+    distance = data[CONF_RADIUS]
+
+    # Check that the provided latitude/longitude provide a response
+    try:
+        test_data = await client.observations.latLong(lat, lng, distance=distance)
+        if not test_data:
+            raise InvalidLocation
+    except InvalidKeyError as exc:
+        raise InvalidAuth from exc
+    except AirNowError as exc:
+        raise CannotConnect from exc
+
+    # Validation Succeeded
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for AirNow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            # Set a unique id based on latitude/longitude
+            await self.async_set_unique_id(
+                f"{user_input[CONF_LATITUDE]}-{user_input[CONF_LONGITUDE]}"
+            )
+            self._abort_if_unique_id_configured()
```

Comment:
Please add a test for existing config entry. Use a MockConfigEntry andaadd it to hass.

https://github.com/home-assistant/core/blob/a5cd4efd83a524d72b250a7324cddbf66d5f58db/tests/common.py#L657